### DISCUSSION TITLE
Source Entropy of BinarySource()

### DISCUSSION THREAD

**Discussion by nadh-amar** [NVlabs/sionna#261] [Q&A]
Hi, 
May I know what is the source entropy for BinarySource() function ? I mean if there is any probability of producing one of the bits or something of that sort..?
Or how I can calculate it ? 
I am new to this whole area, please do help. 
Thanks!

**Comment by SebastianCa** [comment]
Hi,
the `BinarySource` generates a random bit sequence with `p(b=0)=p(b=1)=0.5`.
This leads to a [binary entropy](https://en.wikipedia.org/wiki/Binary_entropy_function) `H_b(p)=1`.

Hint: if you need different values for p, you can use the [BinaryMemorylessChannel](https://nvlabs.github.io/sionna/api/channel.discrete.html#binarymemorylesschannel) and feed an all_zero input.

**Reply by nadh-amar** [reply]
Awesome. I was assuming the same value. Thank you for the help and suggestion.
